### ISSUE TITLE
Add `decay` method

### ISSUE THREAD

**Issue by tomerfiliba** [NewbieOrange/topk#3]
Hi @NewbieOrange, I wanted to add a `decay` method to the FSS such that it would allow me to decrease the weight of historical items. Essentially it should be identical to this 

```rust
let factor = 0.8;
let fss2 = FilteredSpaceSaving::new(fss1.k);
for (item, counter) in fss1.iter() {
    fss2.insert(item, ((counter.estimated_count as f64)*factor) as u64);
}
```

But without allocating a new FSS, and keeping the additional alphas of items that haven't made it into the top k.
This is what I've got - what do you think? I'm not sure about not breaking any invariants with this

```rust
    /// Computes in **O(k)** time
    pub fn decay(&mut self, factor: f64) {
        let mut items = vec![];

        for (item, counter) in self.monitored_list.iter() {
            let counter2 = ElementCounter::new(
                (counter.0.estimated_count as f64 * factor) as u64,
                (counter.0.associated_error as f64 * factor) as u64,
            );
            if counter2.estimated_count != 0 {
                items.push((*item, Reverse(counter2)));
            }
        }
        self.monitored_list.clear();
        self.monitored_list.extend(items);

        for alpha in self.alphas.iter_mut() {
            *alpha = (*alpha as f64 * factor) as u64;
        }

        self.count = (self.count as f64 * factor) as u64;
    }
```

**Comment by NewbieOrange** [comment]
I think it should be safe to keep zero-elements in the FSS? (they are still Top-K elements anyways) So we don't need to allocate another Vec and re-init the priority queue.

```rust
    /// Decays the counters, equivalent to multiplying all counters by the factor.
    ///
    /// Computes in **O(1)** time.
    pub fn decay(&mut self, factor: f64) {
        for (_, value) in self.monitored_list.iter_mut() {
            value.0.estimated_count = (value.0.estimated_count as f64 * factor) as u64;
            value.0.associated_error = (value.0.associated_error as f64 * factor) as u64;
        }
        self.alphas.iter_mut().for_each(|x| *x = (*x as f64 * factor) as u64);
        self.count = (self.count as f64 * factor) as u64
    }
```

**Comment by tomerfiliba** [comment]
thanks! do you want me to submit a PR?

**Comment by NewbieOrange** [comment]
> thanks! do you want me to submit a PR?

You are welcomed to submit a PR. It would be better if you could include a simple test case.

**Comment by NewbieOrange** [comment]
Closed by #4
